geodiff/src/drivers/sqliteutils.h

```diff
@@ -20,17 +21,18 @@
 #define GPKG_ENVELOPE_SIZE_MASK 14
 
 class Buffer;
+class Context;
 
 
 class Sqlite3Db
 {
   public:
     Sqlite3Db();
     ~Sqlite3Db();
-    void open( const std::string &filename );
+    void open( const Logger &logger,  const std::string &filename );
```

Comment:
I would pass context everywhere, and use `context->logger()` only in places where you really want to output some error to log. I think in context there may be in the future some additional information that may be needed in general (like the caller program identifier or whatever else) `logger` class is just really private implementation detail in my view